devnet-sdk/system/chain.go

```diff
@@ -65,19 +63,28 @@ type chain struct {
 	clients  *clientManager
 	registry interfaces.ContractsRegistry
 	mu       sync.Mutex
+
+	node Node
```

Comment:
map[string]Node ?

Reply:
I assume that you are mentioning that a `Chain` can consist of multiple `Nodes` and we pair each rpc url for each Node. The key value of `map[string]Node` will be the rpc url. To make this actually usable, `Chain` must contain multiple RPC urls. If this is needed, I can add in the separate PR.

Reply:
After the discussion, we will have several node types added at the post PRs, such as
- L1 EL RPC
- L2 EL RPCs  (can also be proxyd) (current implementation only has this as a `Node` interface)
- L1 beacon node
- L2 rollup node
...

Reply:
Makes sense. OK to leave https://github.com/ethereum-optimism/optimism/issues/14616 open until all of those PRs are in?

Reply:
@teddyknox yes!